src/backend/distributed/operations/repair_shards.c

```diff
@@ -318,6 +322,8 @@ citus_move_shard_placement(PG_FUNCTION_ARGS)
 	ErrorIfMoveUnsupportedTableType(relationId);
 	ErrorIfTargetNodeIsNotSafeToMove(targetNodeName, targetNodePort);
 
+	AcquirePlacementColocationLock(relationId, ExclusiveLock, "move");
+
```

Comment:
seems like we should do this for split as well

Reply:
Yes, realized that. I'm planning to do it in a separate PR. But, maybe a separate commit here would be better, I can push here.

Reply:
added a check and test